src/tools/management/commands/ingest.py

```diff
@@ -0,0 +1,118 @@
+import argparse
+import json
+import sys
+import uuid
+
+from django.core.management.base import BaseCommand, CommandError
+from django.template.defaultfilters import pluralize
+from django.db import connection, transaction
+
+
+from api.models import CollectRecord
+from api.ingest.utils import ingest_benthicpit, ingest_fishbelt
+from api.models import (
+    BENTHICLIT_PROTOCOL,
+    BENTHICPIT_PROTOCOL,
+    BLEACHINGQC_PROTOCOL,
+    FISHBELT_PROTOCOL,
+    HABITATCOMPLEXITY_PROTOCOL,
+)
+
+protocol_ingests = {
+    BENTHICLIT_PROTOCOL: None,
+    BENTHICPIT_PROTOCOL: ingest_benthicpit,
+    FISHBELT_PROTOCOL: ingest_fishbelt,
+    HABITATCOMPLEXITY_PROTOCOL: None,
+    BLEACHINGQC_PROTOCOL: None,
+}
+
+
+class Command(BaseCommand):
+    help = "Ingest collect records from a CSV file."
+
+    def add_arguments(self, parser):
+        parser.add_argument("datafile", nargs=1, type=argparse.FileType("r"))
+        parser.add_argument("project", nargs=1, type=uuid.UUID)
+        parser.add_argument("profile", nargs=1, type=uuid.UUID)
+        parser.add_argument("--protocol", choices=protocol_ingests.keys())
+        parser.add_argument(
+            "--dry-run",
+            action="store_true",
+            help="Runs ingest in a database transaction, then does a rollback.",
+        )
+        parser.add_argument(
+            "--clear-existing",
+            action="store_true",
+            help="Remove existing collect records for protocol before ingesting file",
+        )
+
+    def clear_collect_records(self, project, protocol):
+        sql = """
+            DELETE FROM {table_name}
+            WHERE 
+                project_id='{project}' AND 
+                data->>'protocol' = '{protocol}';
+            """.format(
+            table_name=CollectRecord.objects.model._meta.db_table,
+            project=project,
+            protocol=protocol,
+        )
+        with connection.cursor() as cursor:
+            cursor.execute(sql)
+            return cursor.rowcount
+
+    def handle(
+        self,
+        datafile,
+        project,
+        profile,
+        protocol,
+        dry_run,
+        clear_existing,
+        *args,
+        **options
+    ):
+        datafile = datafile[0]
+        project = project[0]
+        profile = profile[0]
+        verbosity = options["verbosity"]
+
+        _ingest = protocol_ingests.get(protocol)
+
+        if _ingest is None:
+            raise NotImplementedError()
+
+        try:
+            with transaction.atomic():
+                sid = transaction.savepoint()
+                if clear_existing:
+                    self.clear_collect_records(project, protocol)
+                records, errors = _ingest(datafile, project, profile, protocol, dry_run)
```

Comment:
No `protocol` on the relevant functions

Reply:
_ingest is just an abstract variable Line 80, the specific ingest method is fetched based on protocol